omegler.js

```diff
@@ -19,37 +19,48 @@ $(document).ready(function () {
 
     $("#textbtn").click(function start(){
         console.log("[debug] start");
-        chrome.storage.sync.get("text", function (val) {
-            write(val["text"],true);
-        });
+		chrome.storage.sync.get({text:'Hello there!', delay: 0 }, function (val) {
+			write(val.text, val.delay);
+		});
     });
 
     function write(msg, delay) {
+        console.log("[debug] write called delay=" + delay + "// msg=" + msg);
         $(".chatmsg", document).html(msg);
         if ($(".sendbtn", document).is(":disabled")) {
             return setTimeout(function () {
-                write(msg);
+                write(msg, delay);
             }, 200);
         }
         stopper = 0;
         
-        setTimeout(function(){
-            $(".sendbtn", document).click();
-        }, delay || 0);
+     	chrome.storage.sync.get("delay", function (val) {
+        	setTimeout(function(){
+				console.log("[debug] .sendbtn setTimeout called delay=" + val["delay"]);
+				$(".statuslog:first", document).html(new Date() + "<br>" + $(".statuslog:first", document).html());
+				$(".sendbtn", document).click();
+			}, val.delay || 0);
+        });
     }
 
     function reconnect(double){
-        stopper = 1;
-        msgCount = 0;
-        clearInterval(timeout);
-        $(".disconnectbtn", document).click();
-        if (double) {
-            $(".disconnectbtn", document).click();
+        console.log("[debug] reconnect");
+        if (msgCount < 10){
+            stopper = 1;
+            msgCount = 0;
+            clearInterval(timeout);
             $(".disconnectbtn", document).click();
+            if (double) {
+                $(".disconnectbtn", document).click();
+                $(".disconnectbtn", document).click();
+            }
+            chrome.storage.sync.get({text:'Hello there!', delay: 0 }, function (val) {
+                write(val.text, val.delay);
+            });
+        }
+        else{
+            $(".chatmsg", document).html('Save chat log if desired and then Reload page to continue.');
```

Comment:
Forcing the user to reload seems silly, ergo why I introduced the pause button. You simply unpause and hit reconnect manually and it continues working.